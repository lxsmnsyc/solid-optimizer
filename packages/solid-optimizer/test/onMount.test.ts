import { describe, expect, it } from 'vitest';
import compile from './compile';

describe('onMount', () => {
  it('should work for ImportSpecifier', async () => {
    expect(await compile(`
      import { onMount } from 'solid-js';
      
      onMount(() => {
        console.log('Hello World');
      });
    `)).toMatchSnapshot();
  });
  it('should work for aliased ImportSpecifier', async () => {
    expect(await compile(`
      import { onMount as x } from 'solid-js';
      
      x(() => {
        console.log('Hello World');
      });
    `)).toMatchSnapshot();
  });
  it('should work for aliased string ImportSpecifier', async () => {
    expect(await compile(`
      import { 'onMount' as x } from 'solid-js';
      
      x(() => {
        console.log('Hello World');
      });
    `)).toMatchSnapshot();
  });
  it('should work for namespace', async () => {
    expect(await compile(`
      import * as solid from 'solid-js';
      
      solid.onMount(() => {
        console.log('Hello World');
      });
    `)).toMatchSnapshot();
  });
});
