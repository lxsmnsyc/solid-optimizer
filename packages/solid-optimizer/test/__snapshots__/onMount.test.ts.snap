// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`onMount > should work for ImportSpecifier 1`] = `"import { onMount } from 'solid-js';"`;

exports[`onMount > should work for aliased ImportSpecifier 1`] = `"import { onMount as x } from 'solid-js';"`;

exports[`onMount > should work for aliased string ImportSpecifier 1`] = `"import { 'onMount' as x } from 'solid-js';"`;

exports[`onMount > should work for namespace 1`] = `"import * as solid from 'solid-js';"`;
